[Xcode] Avoid switching to WebKit.xcworkspace from a developer-provided workspace

emw-apple opened the pull request:
#### 5276cca3f477b336f546659f1225c9beb299b32f
<pre>
[Xcode] Avoid switching to WebKit.xcworkspace from a developer-provided workspace
<a href="https://bugs.webkit.org/show_bug.cgi?id=243322">https://bugs.webkit.org/show_bug.cgi?id=243322</a>
rdar://97614880

Reviewed by Alexey Proskuryakov.

Remove the WORKSPACE_PATH in the top-level Makefile, to prevent `make`
from overwriting a workspace set by set-webkit-configuration. This
avoids unnecessary full rebuilds when WebKit is built from an internal
workspace and then `make` is run from the root of the repo. Open-source
builds will still fall back to this workspace.

For added clarity, especially when building WebKit in other workspaces,
rename the full-stack schemes:

    All Tools.xcscheme =&gt; All WebKit Tools.xcscheme
    All Modules.xcscheme =&gt; Everything up to WebKit + Tools.xcscheme
    All Source.xcscheme =&gt; Everything up to WebKit.xcscheme

* Introduction.md:
* Makefile:
* Source/Makefile:
* Tools/Makefile:
* Tools/Scripts/build-webkit:
* WebKit.xcworkspace/xcshareddata/xcschemes/All WebKit Tools.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Tools.xcscheme.
* WebKit.xcworkspace/xcshareddata/xcschemes/Everything up to WebKit + Tools.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Modules.xcscheme.
* WebKit.xcworkspace/xcshareddata/xcschemes/Everything up to WebKit.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Source.xcscheme.

Canonical link: <a href="https://commits.webkit.org/252933@main">https://commits.webkit.org/252933@main</a>
</pre>

emw-apple commented on Introduction.md:
```diff
@@ -1738,7 +1738,7 @@ FIXME: Write how to investigate a test failure.
 
 The easiest way to debug a layout test is with WebKitTestRunner or DumpRenderTree.
-In Product > Scheme, select “All Source”.
+In Product > Scheme, select “Everything up to WebKit + Tools”.
```
This is intentionally a correction, not a rename. The "All Source" scheme wouldn't build WebKitTestRunner or DumpRenderTree, so I don't see how it could be used to run them unless you had already built them first.

aproskuryakov replied:
That wasn't necessary in most cases, as `run-webkit-tests` compiles them. Could cause trouble when running the tools directly, without the script.

emw-apple replied:
From the screenshot included here, it looks like DumpRenderTree used to be part of "All Sources":

![](https://github.com/WebKit/WebKit/raw/main/resources/xcode-build-settings-for-run.png)

IIUC, these directions aren't telling the reader to use `run-webkit-tests` at all. Regardless, it would be good to go through this workflow again, verify it works, and take new screenshots.

emw-apple committed:
[Xcode] Avoid switching to WebKit.xcworkspace from a developer-provided workspace
https://bugs.webkit.org/show_bug.cgi?id=243322
rdar://97614880

Reviewed by Alexey Proskuryakov.

Remove the WORKSPACE_PATH in the top-level Makefile, to prevent `make`
from overwriting a workspace set by set-webkit-configuration. This
avoids unnecessary full rebuilds when WebKit is built from an internal
workspace and then `make` is run from the root of the repo. Open-source
builds will still fall back to this workspace.

For added clarity, especially when building WebKit in other workspaces,
rename the full-stack schemes:

    All Tools.xcscheme => All WebKit Tools.xcscheme
    All Modules.xcscheme => Everything up to WebKit + Tools.xcscheme
    All Source.xcscheme => Everything up to WebKit.xcscheme

* Introduction.md:
* Makefile:
* Source/Makefile:
* Tools/Makefile:
* Tools/Scripts/build-webkit:
* WebKit.xcworkspace/xcshareddata/xcschemes/All WebKit Tools.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Tools.xcscheme.
* WebKit.xcworkspace/xcshareddata/xcschemes/Everything up to WebKit + Tools.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Modules.xcscheme.
* WebKit.xcworkspace/xcshareddata/xcschemes/Everything up to WebKit.xcscheme: Renamed from WebKit.xcworkspace/xcshareddata/xcschemes/All Source.xcscheme.

Canonical link: https://commits.webkit.org/252933@main
